jib-core/src/main/java/com/google/cloud/tools/jib/api/JibContainerBuilder.java

```diff
@@ -119,6 +119,32 @@ public JibContainerBuilder addLayer(List<Path> files, AbsoluteUnixPath pathInCon
     return addLayer(layerConfigurationBuilder.build());
   }
 
+  /**
+   * Adds a new layer to the container with {@code files} as the source files and {@code
+   * pathInContainer} as the path to copy the source files to in the container file system.
+   *
+   * @param files the source files to copy to a new layer in the container
+   * @param pathInContainer the path in the container file system corresponding to the {@code
+   *     sourceFile}
+   * @return this
+   * @throws IOException if an exception occurred when recursively listing any directories
```

Comment:
How about adding

`@throws IllegalArgumentException if {@code pathInContainer} is not an absolute Unix-style path`